### DISCUSSION TITLE
RTE (Extensions and Custom Modules)

### DISCUSSION THREAD

**Discussion by VikingDadMedic** [mantinedev/mantine#1258] [Q&A] [answered]
What is the proper procedure to add [Extensions](https://quilljs.com/docs/api/#extension), [Deltas](https://quilljs.com/docs/delta/), and [Modules](https://quilljs.com/docs/modules/#extending) for the Mantine RTE? 

I saw this was referenced in [Discussion](https://github.com/mantinedev/mantine/discussions/742) #742 but I don't see any reference in the docs as to how. 

My thinking is the entry point for adding them is [RichTextEditor.tsx ](https://github.com/mantinedev/mantine/blob/cac3ac052c8bbe3abb0a5871d3d8a869bcadedb1/src/mantine-rte/src/components/RichTextEditor/RichTextEditor.tsx) and then here `mantine/src/mantine-rte/src/index.ts` ([link](https://github.com/mantinedev/mantine/blob/master/src/mantine-rte/src/index.ts)). Would that be correct reasoning, or is there more to it? Could one basically follow **[this commit](https://github.com/mantinedev/mantine/commit/02323b2ac25764e3a976f0d6da66fa24dbf4c9e8)** pattern with predictably similar success?

@rtivital  ➔ Any walkthrough, docs, examples?

**Comment by rtivital** [comment, marked as answer]
`@mantine/rte` was never meant to be extensible or customizable. I've made it as a simple solution for my projects that require simple html content modifications. If you need anything more, you should use react-quill or other rte library to build your own editor.

**Reply by organicolives** [reply]
I just ran into an odd issue where RTE freezes my React app when I set onChange prop to setText function. The error wasn't happening even an hour ago but now my app freezes after typing a single letter in the RTE text box. I'm not sure what's causing it...

**Reply by organicolives** [reply]
const [text, updateText] = useState(''); //my useState Hook

<RichTextEditor value={text} onChange={updateText} /> //How I create the element in my component
